cli/src/main/java/com/devonfw/tools/ide/tool/ide/IdeToolCommandlet.java

```diff
@@ -1,32 +1,25 @@
 package com.devonfw.tools.ide.tool.ide;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Set;
-import java.util.stream.Stream;
-
 import com.devonfw.tools.ide.cli.CliException;
 import com.devonfw.tools.ide.context.IdeContext;
 import com.devonfw.tools.ide.io.FileAccess;
-import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.PluginBasedCommandlet;
 import com.devonfw.tools.ide.tool.ToolCommandlet;
 import com.devonfw.tools.ide.tool.eclipse.Eclipse;
 import com.devonfw.tools.ide.tool.intellij.Intellij;
 import com.devonfw.tools.ide.tool.vscode.Vscode;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
 /**
  * {@link ToolCommandlet} for an IDE (integrated development environment) such as {@link Eclipse}, {@link Vscode}, or
  * {@link Intellij}.
  */
-public abstract class IdeToolCommandlet extends LocalToolCommandlet {
-
-  private Map<String, PluginDescriptor> pluginsMap;
+public abstract class IdeToolCommandlet extends PluginBasedCommandlet {
 
   private Collection<PluginDescriptor> configuredPlugins;
```

Comment:
This was also moved so it needs to be deleted here (otherwise you have it twice)

Reply:
Seems to apply for many other methods - please move code instead of copying it to the super-class without removing it in the sub-class what is causing duplicates.